sdk/src/wallet/migration/chrysalis.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str::FromStr;
+use std::{
+    collections::{HashMap, HashSet},
+    convert::TryInto,
+    io::Read,
+    path::Path,
+};
+
+use crypto::{
+    ciphers::{chacha::XChaCha20Poly1305, traits::Aead},
+    macs::hmac::HMAC_SHA512,
+};
+use rocksdb::{IteratorMode, Options, DB};
+use serde::Serialize;
+use serde_json::Value;
+use zeroize::Zeroizing;
+
+use crate::{
+    client::{constants::IOTA_COIN_TYPE, Password},
+    types::block::address::Bech32Address,
+    wallet::{
+        migration::{MigrationVersion, MIGRATION_VERSION_KEY},
+        storage::constants::{
+            ACCOUNTS_INDEXATION_KEY, ACCOUNT_INDEXATION_KEY, CHRYSALIS_STORAGE_KEY, SECRET_MANAGER_KEY,
+            WALLET_INDEXATION_KEY,
+        },
+        Error, Result,
+    },
+};
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct AccountAddress {
+    address: Bech32Address,
+    key_index: u32,
+    internal: bool,
+    used: bool,
+}
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct AccountDetailsDto {
+    pub(crate) index: u32,
+    coin_type: u32,
+    alias: String,
+    public_addresses: Vec<AccountAddress>,
+    internal_addresses: Vec<AccountAddress>,
+    addresses_with_unspent_outputs: Vec<Value>,
+    outputs: HashMap<String, Value>,
+    locked_outputs: HashSet<String>,
+    unspent_outputs: HashMap<String, Value>,
+    transactions: HashMap<String, Value>,
+    pending_transactions: HashSet<String>,
+    incoming_transactions: HashMap<String, Value>,
+    native_token_foundries: HashMap<String, Value>,
+}
+
+pub async fn migrate_db_chrysalis_to_stardust(storage_path: String, password: Option<Password>) -> Result<()> {
```

Comment:
Take an impl Into<Path> ?

Reply:
That's not `Send` and then it returns an error

Reply:
You can just add a Send bound, that's what we usually do when we use an impl Into

Reply:
Also fine to keep like this, was a minor suggestion